File: src/cmcandy/C_Language_Answers/_0026.cpp
#include <iostream>
#include <map>
#include <vector>

using namespace::std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int cur=0;
        map<int,int>m;
        for (int i = 0; i < nums.size(); i++)
        {
            if (m.find(nums[i])==m.end())
            {
                //如果找不到,则加入
                m[nums[i]]=1;
                nums[cur++] = nums[i];
            }
        }
        return cur;
    }
};

File: src/cmcandy/Python_language_Answers/_0026.py
from typing import List
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        length = len(nums)
        cur = 0
        map = {}
        for index in range(length):
            if map.get(nums[index]) is None:
                map[nums[index]] = 1
                nums[cur] = nums[index]
                cur += 1
        return cur             
            
